### ISSUE TITLE
When a tunnel port is already in use the error message logged in the client grows with each connection attempt

### ISSUE THREAD

**Issue by chibbert** [erebe/wstunnel#242]
**Describe the bug**
We've started looking to use wstunnel for our use case and love what it can do, thanks for a great project!

When creating a simple local setup to test wstunnel for our use case (reverse tunnel) I noticed that when I tested the behaviour of two clients connecting using the same tunnelling port then the INFO logs start getting longer and longer 

```2024-03-14T11:25:44.517166Z  INFO tunnel{id="018e3ca3-a8e6-7d39-862c-4f16e66d6648" remote="[::]:9001"}:tunnel{id="018e3ca3-b0bb-731b-a5f2-d83debff77a5" remote="[::]:9001"}:tunnel{id="018e3ca3-b890-745c-adb8-547a9fabea68" remote="[::]:9001"}:tunnel{id="018e3ca3-c065-705d-9f8c-8ae55cf8eadc" remote="[::]:9001"}:tunnel{id="018e3ca3-c839-7e08-b306-53f9c015c3bf" remote="[::]:9001"}:<SNIPPED many repetitions>:tunnel{id="018e3cb5-a215-7106-99de-66569a2467b5" remote="[::]:9001"}:tunnel{id="018e3cb5-a9f8-7efd-a063-f5b743bbd3f8" remote="[::]:9001"}:tunnel{id="018e3cb5-b1da-7cab-bb44-f1f1e88d7ccc" remote="[::]:9001"}:tunnel{id="018e3cb5-b9be-7c86-b72b-2f1a8b34e0a4" remote="[::]:9001"}:tunnel{id="018e3cb5-c1a2-7ec2-82ee-1bdf46e7403f" remote="[::]:9001"}:tunnel{id="018e3cb5-c986-7276-8a54-0e07d50ef4f7" remote="[::]:9001"}:tunnel{id="018e3cb5-d16a-7d56-b382-c2c392cdf0a3" remote="[::]:9001"}:tunnel{id="018e3cb5-d94e-74e8-a5e6-5f595ca0785b" remote="[::]:9001"}:tunnel{id="018e3cb5-e132-7dea-b654-1ea3eb7503d0" remote="[::]:9001"}:tunnel{id="018e3cb5-e915-7a6f-aabf-e30bf10e28fe" remote="[::]:9001"}:tunnel{id="018e3cb5-f0f9-790a-bc62-70f082fe9b3c" remote="[::]:9001"}:tunnel{id="018e3cb5-f8de-7705-9277-d2a4bd5bb341" remote="[::]:9001"}:tunnel{id="018e3cb6-00c2-78d5-a555-a4be372d62ae" remote="[::]:9001"}: wstunnel::tls: Doing TLS handshake using sni IpAddress(127.0.0.1) with the server 127.0.0.1:8443
```

**To Reproduce**

```
# Start server end of wstunnel:
wstunnel server wss://[::]:8443

# Start client end of wstunnel:
wstunnel client -R tcp://[::]:9001:localhost:8001 wss://127.0.0.1:8443

# Start another client using same port 9001 of wstunnel:
wstunnel client -R tcp://[::]:9001:localhost:8002 wss://127.0.0.1:8443

# You will see the second client logs repeatedly trying to connect, but each time it fails, the error message that ends with:
.... wstunnel::tcp: Cannot connect to tcp endpoint 127.0.0.1:8443 reason Connection refused (os error 111)

grows and it looks like the details used for each connection attempt are appended to a list
```

**Expected behavior**
The connection error should only show the tunnel details for the current connection attempt that just failed.

**Screenshots**
![image](https://github.com/erebe/wstunnel/assets/5433799/177ea670-72c2-40d7-b049-291a4474dcdf)

**Desktop (please complete the following information):**
 - OS: Ubuntu 
 - Browser: Curl (version 7.68.0)
 - Version: Downloaded version: https://github.com/erebe/wstunnel/releases/download/v9.2.4/wstunnel_9.2.4_linux_amd64.tar.gz

**Additional context**

**Comment by erebe** [comment]
Thanks for reporting, I am going to fix the issue.

**Comment by erebe** [comment]
Fixed in release 9.2.4 https://github.com/erebe/wstunnel/releases/tag/v9.2.4

**Comment by JamesTGrant** [comment]
Hi @erebe, me again! I have noticed this issue is also present in the server. I have attached the logs from the server
[growing.txt](https://github.com/user-attachments/files/17061898/growing.txt)
